### PR TITLE
Android/Contextual: Reset sheet when contextual chat has been deleted

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1202552961248957/task/1216752685086432?focus=true
Tech Design URL (if applicable): 

### Description
 - When the contextual Duck.ai sheet reopens onto a stored session whose chat is no longer in history (deleted while the sheet was away), it now resets to the input state instead of restoring WEBVIEW mode on a chat that no longer loads.
  - The check runs synchronously at session-restore time in onSheetOpened and reopenWebViewState (isStoredChatMissingFromHistory), so a dead chat URL is never loaded into the WebView.
  - Removed the live observeCurrentChatDeletion observer (plus handleLoadedChatDeleted and lastSeenChatId) in favor of the restore-time check — the reset now happens on next open rather than live. There are no real scenarios when this could happen (aside from chat sync - but that is  very edge case)

### Steps to test this PR
Test: 
- https://app.asana.com/1/137249556945/task/1215727998445071/comment/1215764372557473?focus=true to ensure this scenario is still fixed - original observe was added for this scenario
- https://app.asana.com/1/137249556945/project/1202552961248957/task/1216662041783905?focus=true - issue to be fixed

---

> [!NOTE]
> **Medium Risk**
> Touches contextual session restore and WebView load paths; UX differs from live reset (deletion while sheet is open is deferred until reopen), and a brand-new chat not yet in history could be treated as missing if opened before persistence.
> 
> **Overview**
> **Replaces live “chat deleted elsewhere” handling with a restore-time check** so the contextual sheet does not reopen into **WEBVIEW** on a stored URL whose `chatID` is no longer in chat history.
> 
> The flow observer (`observeCurrentChatDeletion`, `lastSeenChatId`, `handleLoadedChatDeleted`) is removed. **`isStoredChatMissingFromHistory`** compares the stored URL’s chat id to a one-shot read of `chatHistoryRepository.observeChats()` and, when missing, calls **`renderNewChatState()`** from **`onSheetOpened`** and **`reopenWebViewState`** instead of loading the dead URL. Session reuse now requires both a valid timeout and the chat still being in history.
> 
> **Behavior change:** reset happens on the **next open/reopen**, not while the sheet stays visible with a loaded chat. The deletion check is **not** gated on `contextualSheetImprovements` (tests cover improvements on and off). Unit tests were narrowed to open-time scenarios; restore tests seed history when reusing stored URLs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 195e8ebeee9dff76f2642996d8ef7e7b8d9bf01b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by karlenDimla** [comment]
* **#9259** <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9259?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9259?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualViewModel.kt [hunk 1]
```diff
@@ -217,35 +217,9 @@ class DuckChatContextualViewModel @Inject constructor(
             .launchIn(viewModelScope)
     }
 
-    // Last chat id we confirmed exists in history. A brand-new chat sets _chatId (from the loaded
-    // URL) before it is persisted, so "absent from history" only means "deleted" once we've actually
-    // seen it present — this field is what distinguishes the two.
-    private var lastSeenChatId: String? = null
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualViewModel.kt:223]: Deferring for now. 
  
The gap is real in theory as the `isStoredChatMissingFromHistory` can't tell a not-yet-persisted chat from a deleted one, but it isn't easily reachable in practice. 
  
The real issue is that an in-progress chat isn't restored on reopen at all but fixing that means reworking when the session is persisted/restored, which is a bigger change. Deferring and folding it into that work rather than adding a targeted guard now.

**Review by YoussefKeyrouz — APPROVED** [review]
Tested and works well. Thanks
